admin/src/AuthProvider.ts

```diff
@@ -0,0 +1,74 @@
+import { addRefreshAuthToAuthProvider, AuthProvider } from "react-admin";
+import { jwtDecode } from "jwt-decode";
+import {
+	authenticate,
+	logout,
+	refreshAuthentication,
+} from "../../web/app/src/api/accounts.ts";
+import { dataProvider } from "./DataProvider.ts";
+
+export const authProvider: AuthProvider = addRefreshAuthToAuthProvider(
+	{
+		async login({ username, password }) {
+			try {
+				const token = await authenticate(username, password);
+				localStorage.setItem("authentication", token);
+
+				// Check authentication by running a test request.
+				await dataProvider.getList("accounts", {
+					meta: { limit: 1 },
+					pagination: {
+						page: 1,
+						perPage: 1,
+					},
+					filter: {},
+				});
```

Comment:
I don't think it's necessary. The `authenticate` call should return an error if you're not authorized. Maybe by checking we're loging in the admin (URL of the request)?